src/rmet/metrics.py

```diff
@@ -150,6 +151,38 @@ def hitrate(logits: torch.Tensor, targets: torch.Tensor, k: int = 10, logits_are
     return recall
 
 
+def mean_average_precision(logits: torch.Tensor, targets: torch.Tensor, k: int = 10,
+                           logits_are_top_indices: bool = False):
+    """
+    Computes the mean_average_precision@k (MAP@k) for items.
+    In short, it combines precision values at all possible recall levels.
+
+    :param logits: prediction matrix about item relevance
+    :param targets: 0/1 matrix encoding true item relevance, same shape as logits
+    :param k: top k items to consider
+    :param logits_are_top_indices: whether logits are already top-k sorted indices
+    """
+    if k <= 0:
+        raise ValueError("k is required to be positive!")
+
+    top_indices = _get_top_k(logits, k, logits_are_top_indices, sorted=False)  # (n_samples, k)
```

Comment:
The rest of the code assumes that `top_indices` is sorted in descending order of logits, because it iterates over different cutoffs`ki`. If it is like that, then everything looks fine :)

Reply:
Great catch! I'll update this to have the indices sorted :)